emukit/core/optimization/random_search_acquisition_optimizer.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import Tuple
+
+import numpy as np
+from GPyOpt.optimization.acquisition_optimizer import ContextManager
+
+from .. import ParameterSpace
+from ..acquisition import Acquisition
+from ..optimization.acquisition_optimizer import AcquisitionOptimizerBase
+
+_log = logging.getLogger(__name__)
+
+
+class RandomSearchAcquisitionOptimizer(AcquisitionOptimizerBase):
+    """ Optimizes the acquisition function by evaluating at random points."""
+    def __init__(self, space: ParameterSpace, num_samples: int) -> None:
+        """
+        :param space: The parameter space spanning the search problem.
+        :param num_samples: Number of random sampled points which are evaluated per optimization.
+        """
+        self.space = space
+        self.gpyopt_space = space.convert_to_gpyopt_design_space()
+        self.num_samples = num_samples
+
+    def optimize(self, acquisition: Acquisition, context: dict = None) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Optimizes the acquisition function, taking into account gradients if it supports them
+        :param acquisition: The acquisition function to be optimized
+        :param context: Optimization context.
+                        Determines whether any variable values should be fixed during the optimization
+        :return: Tuple of (location of maximum, acquisition value at maximizer)
+        """
+        if context is not None:
```

Comment:
This bit with context is repeated across all optimizers. Perhaps move it to base class, thus making the base class an actual class and making my previous comment invalid?

Reply:
Please check my latest commit. I moved most of the context handling to the base class. This also removes the direct dependency on GPyOpt from most optimizer files (regarding #140 ).

Reply:
Neat, thanks! It broke the build though, but should be a simple enough fix.

Reply:
Sorry, I forgot adding the context manager file. It simply wraps the GPyOpt one and should make it straight forward replacing by an implementation without GPyOpt in future PRs.